pkgs/applications/video/media-downloader/default.nix

```
@@ -0,0 +1,37 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, qt5
+, ffmpeg-full
+, aria2
+, yt-dlp
+, python3
+}:
+
+stdenv.mkDerivation rec {
+  pname = "media-downloader";
+  version = "2.4.0";
+
+  src = fetchFromGitHub {
+    owner = "mhogomchungu";
+    repo = pname;
+    rev = "${version}";
+    sha256 = "sha256-EyfhomwBtdAt6HGRwnpiijm2D1LfaCAoG5qk3orDG98=";
+  };
+
+  nativeBuildInputs = [ cmake qt5.wrapQtAppsHook ];
+
+  preFixup = ''
+    qtWrapperArgs+=(
+    --prefix PATH : "${lib.makeBinPath [ ffmpeg-full aria2 yt-dlp python3 ]}"
+    )
+  '';
+
```

You could also add this to get automatic updates when a new version is tagged.

(needs `nix-update-script` in arguments too)
```suggestion

  passthru.updateScript = nix-update-script {
    attrPath = pname;
  };
```

---

Darwin marked, maybe consider add `nix-update-script` next time.